src/ir/context.rs

```diff
@@ -505,9 +505,15 @@ impl<'ctx> BindgenContext<'ctx> {
                 found_invalid_template_ref = true;
             }
             if c.kind() == CXCursor_TypeRef {
+                // The `with_id` id will potentially end up unused if we find we
+                // already have an item (with its own distinct `ItemId`) for
+                // this type, so if we pass `with_id` as the parent, it is
+                // potentially a dangling reference. Instead, use the canonical
+                // template declaration as the parent. It is already parsed and
+                // has a known-resolvable `ItemId`.
```

Comment:
I think this comment is not as accurate as it should. This ends up unused if we peek the early return in line 555 (the one with the `error!` thing, it's not about finding other type that is the same.